cmds/live.c

```diff
@@ -438,12 +438,13 @@ int command_live(int argc, char *argv[], struct uftrace_opts *opts)
 
 			if (fd < 0)
 				pr_err("cannot create temp name");
-			tmp_dirname = template;
 		}
 
 		close(fd);
-		unlink(tmp_dirname);
 
+		unlink(template);
+
+		snprintf(tmp_dirname, strlen(template), "%s", template);
```

Comment:
You should use `sizeof(tmp_dirname)` instead of `strlen(template)`, otherwise it might not set the last NUL byte.

Reply:
Thank you @namhyung. good advice! 👍